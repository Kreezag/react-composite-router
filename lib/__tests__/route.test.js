/* eslint-env browser, jest */


import { getRoutesByName } from '../route';
import { route } from '../index';


describe('route', () => {
  it('should create some route', () => {
    const someRoute = route.create('some', {
      url: '/some/:id',
      params: {
        id: String(-1),
        _additional: true
      }
    });

    someRoute.create('some.child', {
      url: '/child?some&param'
    });

    expect(getRoutesByName()).toMatchSnapshot();
  });

  it('should create some route', () => {
    route.create('second', {
      url: '/second'
    });

    expect(getRoutesByName()).toMatchSnapshot();
  });
});
