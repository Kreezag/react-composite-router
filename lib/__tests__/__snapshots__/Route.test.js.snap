// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Route should calc new location 1`] = `
Object {
  "hash": "",
  "pathname": "/hello/123",
  "search": "",
}
`;

exports[`Route should parse location 1`] = `
Object {
  "hello": "123",
  "id": "123",
}
`;
